### PR TITLE
(Hacky) People You May Know

### PR DESCRIPTION
**Pull request by leogrim**
@stkem @eishay riding the feed user scoring endpoint, will have to make its own at some point, but that will get us started to build the front-end.  
@yipingliao I can tweak the API on demand. Let's also discuss whether we should have a single or batch endpoint for crossing out recommendations.

### CODE REVIEW

**Comment by eishay** [comment]
thanks @leogrim !

File: kifi-backend/modules/abook/app/com/keepit/abook/commanders/FriendRecommendationCommander.scala [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    db.readWrite { implicit session =>
+      irrelevantUserIds.foreach(friendRecommendationRepo.recordIrrelevantRecommendation(userId, _))
+    }
+  }
+
+  def getRecommendedUsers(userId: Id[User], page: Int, pageSize: Int): Future[Seq[Id[User]]] = {
+    val futureFriendships = graph.getUserFriendships(userId, bePatient = false)
+    val futureFriends = shoebox.getFriends(userId)
+    val futureFriendRequests = shoebox.getFriendRequestsBySender(userId)
+    val rejectedRecommendations = db.readOnlyMaster { implicit session =>
```
Comment (eishay) [inline kifi-backend/modules/abook/app/com/keepit/abook/commanders/FriendRecommendationCommander.scala:30]: no caching there, why not use the replica?

File: kifi-backend/modules/abook/app/com/keepit/abook/model/FriendRecommendation.scala [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+@Singleton
+class FriendRecommendationRepoImpl @Inject() (
+    val db: DataBaseComponent,
+    val clock: Clock) extends DbRepo[FriendRecommendation] with FriendRecommendationRepo {
+
+  import db.Driver.simple._
+
+  type RepoImpl = FriendRecommendationTable
+  class FriendRecommendationTable(tag: Tag) extends RepoTable[FriendRecommendation](db, tag, "friend_recommendation") {
```
Comment (eishay) [inline kifi-backend/modules/abook/app/com/keepit/abook/model/FriendRecommendation.scala:36]: +1 for a repo not in shoebox

File: kifi-backend/modules/abook/app/com/keepit/abook/model/FriendRecommendation.scala [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  initTable()
+
+  override def deleteCache(emailAccount: FriendRecommendation)(implicit session: RSession): Unit = {}
+  override def invalidateCache(emailAccount: FriendRecommendation)(implicit session: RSession): Unit = {}
+
+  private val compiledGetByUserAndFriend = Compiled { (userId: Column[Id[User]], friendId: Column[Id[User]]) =>
+    for (row <- rows if row.userId === userId && row.friendId === friendId) yield row
+  }
+
+  def internFriendRecommendation(userId: Id[User], friendId: Id[User], irrelevant: Boolean)(implicit session: RWSession): FriendRecommendation = {
```
Comment (eishay) [inline kifi-backend/modules/abook/app/com/keepit/abook/model/FriendRecommendation.scala]: is the `internFriendRecommendation` used outside this repo?

**Comment by eishay** [comment]
no testing at all?

File: kifi-backend/modules/abook/app/com/keepit/abook/model/FriendRecommendation.scala (hunk 3 above)
Comment (leogrim) [inline kifi-backend/modules/abook/app/com/keepit/abook/model/FriendRecommendation.scala]: No, I meant to expose it as a primary constructor given the unique constraint on the table. But I can make it private now.

**Comment by leogrim** [comment]
I was hoping that adding "hacky" to the title of the pull request would prevent me from having to write tests :) Seriously though, there's not much room for unit tests, this is basically an aggregation of several existing endpoints. I'll add a short test for the repo.

**Comment by eishay** [comment]
maybe testing `recordIrrelevantRecommendation` that calls `internFriendRecommendation`. the `internFriendRecommendation` seems not trivial and you want to at least test that you can persist and recover objects to/from the db in case you had a typo in the db patch (`210.sql`) so it didn't match the repo column definitions.

**Comment by leogrim** [comment]
Done, that's right, always good to test the patch :+1:

**Comment by eishay** [comment]
thanks!

**Pull request merged**